oolab/src/main/java/agh/ics/oop/Animal.java

```diff
@@ -36,13 +30,18 @@ public String toString() {
         };
     }
 
-    public boolean isAt(Vector2d position) {
-        return Objects.equals(this.position, position);
-    }
-
     private void modifyPosition(Vector2d deltaPosition) {
         Vector2d newPosition = position.add(deltaPosition);
-        if (map.canMoveTo(newPosition) && !map.isOccupied(newPosition)) {
+        AbstractWorldMapElement objectFound = (AbstractWorldMapElement) map.objectAt(newPosition);
+        if (!map.canMoveTo(newPosition)){
+            return;
+        }
+        if (objectFound instanceof Grass){
+            map.deleteObject(objectFound);
+            position = newPosition;
+            ((GrassField)map).placeOneField();
```

Comment:
tu wchodzą niezbyt ładne sprawdzania typu, co od razu sugeruje, że może dałoby się to lepiej poukładać. Zaszyłbym całą tę logikę zjadania trawy w mapie, i to konkretnie w mapie typu `GrassField`. Wymagałoby to ofc dodawania do interfejsu mapy jakiejś metody typu `moved(Animal)` i w przypadku prostokątnej mapy taka metoda by nic nie robiła, ale i tak jest to rozwiązanie, które rozplątuje kod.